### PR TITLE
Early stopping validation set

### LINKED ISSUE #98: Early stopping in NN training
Often neural networks reach a point in their training where they stop improving and may even get worse. We can detect that this is happening by looking at the validation loss/scores. We could have these models stop their training early to avoid wasted effort and get results sooner. We could have a parameter that indicates if a model has not improved in the last k epochs of training, it should stop.

### PR DESCRIPTION
**Pull request by bjschoenfeld**
closes #98

modifies the data splitting function to allow for a test size to be a float between 0 and 1, indicating a ratio of the train data to use as test data

### CODE REVIEW

File: dna/models/base_models.py [hunk 1]
```diff
@@ -65,12 +76,18 @@ def fit(
...
         validation_data_loader = None
         min_loss_score = np.inf
+        min_loss_epoch = -1
         if validation_data is not None:
             validation_data_loader = self._get_data_loader(validation_data, batch_size, drop_last=False, shuffle=False)
 
+        if patience < 1:
+            patience = np.inf
+
         for e in range(n_epochs):
```
Comment (orionw) [inline dna/models/base_models.py]: It took me a second to realize what e was (error / epochs) but maybe it's just been a long day haha.  Not that you did that change, just saying.

Comment (bjschoenfeld) [reply dna/models/base_models.py]: Haha, perhaps that's not the best variable name since `e` is often used for errors.

**Comment by orionw** [comment]
Note: just re-running the one job that was failing caused it to work.  Not sure why.  We can dig into it more if it happens again.
